management/server/sql_store.go

```diff
@@ -274,9 +274,23 @@ func (s *SqlStore) GetInstallationID() string {
 func (s *SqlStore) SavePeerStatus(accountID, peerID string, peerStatus nbpeer.PeerStatus) error {
 	var peerCopy nbpeer.Peer
 	peerCopy.Status = &peerStatus
+
+	peerStatusJson, err := json.Marshal(peerCopy.Status)
+	if err != nil {
+		log.Errorf("cannot marshal peer %s status", peerID)
+		return err
+	}
+
+	parsedPeerStatus := make(map[string]interface{})
+	err = json.Unmarshal(peerStatusJson, &parsedPeerStatus)
+	if err != nil {
+		log.Errorf("cannot unmarshal peer %s status", peerID)
+		return err
+	}
+
 	result := s.db.Model(&nbpeer.Peer{}).
 		Where("account_id = ? AND id = ?", accountID, peerID).
-		Updates(peerCopy)
+		Updates(parsedPeerStatus)
```

Comment:
Thank you for your work on fixing this issue. Your approach using JSON marshaling and unmarshaling works well. However, we can simplify it further.

Instead of marshaling and unmarshaling, we can directly select and update only the fields we need. By passing the peerCopy struct and ignoring the fields we don't want to update, we can streamline the process and avoid the additional steps of marshaling and unmarshaling

```
	fieldsToUpdate := []string{
		"peer_status_last_seen", "peer_status_connected",
		"peer_status_login_expired", "peer_status_required_approval",
	}
	result := s.db.Model(&nbpeer.Peer{}).Select(fieldsToUpdate).
		Where("account_id = ? AND id = ?", accountID, peerID).
		Updates(&peerCopy)
	if result.Error != nil {
		return result.Error
	}

```